sonarts-core/src/utils/sonar-analysis.ts

```diff
@@ -82,20 +86,17 @@ export class IssueLocation {
   }
 }
 
+export function getIssueLocationAtNode(node: ts.Node, message?: string) {
+  return new IssueLocation(node.getStart(), node.getEnd(), node.getSourceFile(), message);
+}
+
 export class SonarIssue extends tslint.RuleFailure {
   private cost?: number;
   public readonly primaryLocation: IssueLocation;
   private secondaryLocations: IssueLocation[] = [];
 
   public constructor(primaryLocation: IssueLocation, ruleName: string) {
-    super(
-      primaryLocation.node.getSourceFile(),
-      primaryLocation.node.getStart(),
-      primaryLocation.lastNode.getEnd(),
-      primaryLocation.message!,
-      ruleName,
-    );
-
+    super(primaryLocation.sourceFile, primaryLocation.start, primaryLocation.end, primaryLocation.message!, ruleName);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Forbidden non null assertion ([typescript:S2966](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Fno-commented-code&issues=AWC7g8QeqDrCo5ysghkQ&open=AWC7g8QeqDrCo5ysghkQ)

[ISSUE_KEY]: AWC7g8QeqDrCo5ysghkQ